ZenPacks/zenoss/Microsoft/Windows/datasources/PerfmonDataSource.py

```diff
@@ -798,37 +798,49 @@ def cleanup(self, config):
 
     def _errorMsgCheck(self, errorMessage):
         """Check error message and generate appropriate event."""
+        datasources = self.config.datasources
+        plugin_classname = datasources[0].plugin_classname
         if 'Password expired' in errorMessage:
             PERSISTER.add_event(self.config.id, {
                 'device': self.config.id,
                 'severity': ZenEventClasses.Critical,
                 'eventClassKey': 'MW|PasswordExpired',
                 'summary': errorMessage,
-                'ipAddress': self.config.manageIp})
+                'ipAddress': self.config.manageIp,
+                'plugin_classname': plugin_classname,
```

Comment:
Rather than writing out this same change all over the place, what about making a utility method on a base plugin class or something like that to automatically add these generally-useful fields?